pkg/gcc/adaptive_threshold.go

```diff
@@ -0,0 +1,60 @@
+package gcc
+
+import (
+	"time"
+)
+
+type adaptiveThresholdOption func(*adaptiveThreshold)
+
+func setInitialThreshold(t time.Duration) adaptiveThresholdOption {
+	return func(at *adaptiveThreshold) {
+		at.thresh = t
+	}
+}
+
+type adaptiveThreshold struct {
+	thresh              time.Duration
+	overuseCoefficientU float64
+	overuseCoefficientD float64
+	min                 time.Duration
+	max                 time.Duration
+}
+
+func newAdaptiveThreshold(opts ...adaptiveThresholdOption) *adaptiveThreshold {
+	at := &adaptiveThreshold{
+		thresh:              time.Duration(125 * float64(time.Microsecond)),
+		overuseCoefficientU: 0.01,
```

Comment:
Where do these values come from? Would it be possible to link to a RFC/PDF?

Maybe we could write an informational one if not?

Reply:
I made some more changes to the rate controller and adaptive threshold and added some explanations.